pkg/smi/client.go

```diff
@@ -32,7 +30,7 @@ import (
 const kubernetesClientName = "MeshSpec"
 
 // NewMeshSpecClient implements mesh.MeshSpec and creates the Kubernetes client, which retrieves SMI specific CRDs.
-func NewMeshSpecClient(smiKubeConfig *rest.Config, kubeClient kubernetes.Interface, osmNamespace string, namespaceController k8s.NamespaceController, stop chan struct{}) (MeshSpec, error) {
+func NewMeshSpecClient(smiKubeConfig *rest.Config, kubeClient kubernetes.Interface, osmNamespace string, kubernetesClient k8s.Client, stop chan struct{}) (MeshSpec, error) {
```

Comment:
`pkg/kubernetes` should expose an interface instead of k8s.Client type. If `Client` is an interface, we should rename it to something meaningful.

Reply:
done